### PR TITLE
[JENKINS-36193] Raise exception when trying to build a job that is di…

### PR DESCRIPTION
**Pull request by godfath3r**
…sabled.

First time patching Jenkins code, so please be kind. :)

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
👍 Since it's a code copy from the doBuild() method

**Comment by godfath3r** [comment]
I've just pushed commit 0a7a259 for another issue JENKINS-36387, I now know that pull requests on github are branch based... :S 

@oleg-nenashev Could you probably cherry pick commits independently or should I make 2 new PRs for those 2 commits?

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java [hunk 1]
```diff
@@ -215,6 +215,9 @@ public final void doBuildWithParameters(StaplerRequest req, StaplerResponse rsp,
         hudson.model.BuildAuthorizationToken.checkPermission(asJob(), asJob().getAuthToken(), req, rsp);
 
         ParametersDefinitionProperty pp = asJob().getProperty(ParametersDefinitionProperty.class);
+        if (!asJob().isBuildable()) {
+            throw HttpResponses.error(SC_INTERNAL_SERVER_ERROR, new IOException(asJob().getFullName() + " is not buildable!"));
```
Comment (jtnord) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: I'm not entirly convicned that a 500 response code here is correct.
This is not an error in the server - it's an error with the request.

As it is a client error (asking to build a job that is not buildable) it should be a 4xx response.

possibly a [409 Conflict ](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.10)

Comment (godfath3r) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: Your point is correct, 409 response seems correct. I'll change it to that. 

Maybe the same should be also applied on doBuild() method?

**Pull request closed without merging**